internal/weapons/catalyst/hakushin/hakushin.go

```diff
@@ -0,0 +1,47 @@
+package hakushin
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/pkg/core"
+)
+
+func init() {
+	core.RegisterWeaponFunc("hakushin ring", weapon)
+	core.RegisterWeaponFunc("hakushin", weapon)
+}
+
+func weapon(char core.Character, c *core.Core, r int, param map[string]int) string {
+
+	expiry := 0
+	e := 10 + float64(r)*2.5
+	m := make([]float64, core.EndStatType)
+	m[core.ElectroP] = e
+	hrfunc := func(ele core.EleType, key string) func(args ...interface{}) bool {
+		return func(args ...interface{}) bool {
+			if c.ActiveChar != char.CharIndex() {
+				return false
+			}
+			m[ele] = e
+			expiry = c.F + 6*60
+			return false
+		}
+	}
+
+	for _, char := range c.Chars {
+		char.AddMod(core.CharStatMod{
+			Key: "hakushin-passive",
```

Comment:
I think this will cause collisions in naming on buff refresh? Pretty sure you need a separate name for each element.